packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,539 @@
+                                                        w="100%"
+                                                        h="100%"
+                                                    />
+                                                </Box>
+                                                <ItemVariantProvider variant={variant}>
+                                                    <VStack
+                                                        justify="flex-start"
+                                                        minW={0}
+                                                        flex={1}
+                                                        pt={0}
+                                                        align="flex-start"
+                                                    >
+                                                        <Text
+                                                            id={`product-title-${addressKey}`}
+                                                            data-testid={`product-title-${addressKey}`}
+                                                            fontWeight="medium"
+                                                            fontSize={{base: 'sm', md: 'md'}}
+                                                            mb={1}
+                                                            color="gray.900"
+                                                            textAlign="left"
+                                                        >
+                                                            {item.productName}
+                                                        </Text>
+                                                        <Box
+                                                            id={`product-description-${addressKey}`}
+                                                            data-testid={`product-description-${addressKey}`}
+                                                        >
+                                                            <MultiShippingItemAttributes
+                                                                variant={variant}
+                                                                includeQuantity
+                                                            />
+                                                        </Box>
+                                                    </VStack>
+                                                </ItemVariantProvider>
+                                            </HStack>
+                                        </Flex>
+
+                                        {/* Right: Address dropdown and price - only on desktop */}
+                                        <HideOnMobile>
+                                            <VStack
+                                                align="flex-start"
+                                                w={{base: '100%', md: '340px'}}
```

Comment:
Can we find away to remove this magic number 340px? Using flex should be able to help. Or we can try grid.